sbncode/FlashMatch/FlashPredict.hh

```diff
@@ -83,59 +79,207 @@ public:
                     const art::Ptr<recob::PFParticle>& pfp_ptr,
                     const art::ValidHandle<std::vector<recob::PFParticle>>& pfps_h,
                     std::vector<art::Ptr<recob::PFParticle>>& pfp_v) const;
-  double scoreTerm(const double m, const double n,
-                   const double mean, const double spread) const;
-  double scoreTerm(const double m,
-                   const double mean, const double spread) const;
-  bool pfpNeutrinoOnEvent(
+  void updateChargeMetrics(const ChargeMetrics& chargeMetrics);
+  void updateFlashMetrics(const FlashMetrics& flashMetrics);
+  void updateScore(const Score& score);
+  inline double scoreTerm(const double m, const double n,
+                          const double mean, const double spread) const;
+  inline double scoreTerm(const double m,
+                          const double mean, const double spread) const;
+  inline bool pfpNeutrinoOnEvent(
     const art::ValidHandle<std::vector<recob::PFParticle>>& pfps_h) const;
   void copyOpHitsInBeamWindow(
     std::vector<recob::OpHit>& opHits,
     const art::Handle<std::vector<recob::OpHit>>& ophit_h) const;
-  bool getOpHitsInFlash(std::vector<recob::OpHit>& opHits);
-  bool createOpHitsTimeHist(
-    const std::vector<recob::OpHit>& opHits) const;
-  bool findMaxPeak(std::vector<recob::OpHit>& opHits);
+  std::vector<SimpleFlash> makeSimpleFlashes(//SBND overload
+    std::vector<recob::OpHit>& opHits,
+    std::vector<recob::OpHit>& opHitsRght,
+    std::vector<recob::OpHit>& opHitsLeft) const;
+  std::vector<SimpleFlash> makeSimpleFlashes(//ICARUS overload
+    std::vector<recob::OpHit>& opHits) const;
+  bool createOpHitsTimeHist(//SBND overload
+    const std::vector<recob::OpHit>& opHits,
+    std::vector<recob::OpHit>& opHitsRght,
+    std::vector<recob::OpHit>& opHitsLeft,
+    std::unique_ptr<TH1D>& opHitsTimeHist,
+    std::unique_ptr<TH1D>& opHitsTimeHistRght,
+    std::unique_ptr<TH1D>& opHitsTimeHistLeft) const;
+  unsigned createOpHitsTimeHist(//ICARUS overload
+    const std::vector<recob::OpHit>& opHits,
+    std::unique_ptr<TH1D>& opHitsTimeHist) const;
+  bool findSimpleFlashes(
+    std::vector<SimpleFlash>& simpleFlashes,
+    std::vector<recob::OpHit>& opHits,
+    const unsigned ophsInVolume,
+    std::unique_ptr<TH1D>& opHitsTimeHist) const;
+  inline std::string detectorName(const std::string detName) const;
   bool isPDInCryo(const int pdChannel) const;
   bool isSBNDPDRelevant(const int pdChannel,
                         const std::set<unsigned>& tpcWithHits) const;
   unsigned sbndPDinTPC(const int pdChannel) const;
   unsigned icarusPDinTPC(const int pdChannel) const;
+  double wallXWithMaxPE(const OpHitIt opH_beg,
+                        const OpHitIt opH_end) const;
+  std::list<double> wiresXGl() const;
+  double driftDistance() const;
   double flashXGl(const double hypo_x, const double flash_x) const;
-  double driftDistance(const double x) const;
+  double foldXGl(const double x_gl) const;
```

Comment:
This name is pretty cryptic to me, some comments to explain what these functions do a bit more verbosely would help